Update missing GroundDomain output variables and documentation

mitchute opened the pull request:
@mjwitte @Myoldmopar 

Addresses #5321

Please review output variable name changes. What sort of unit test would you recommend for this?

Commit:
Init slabs and basements in right place (before ProcessRDDMDD()).
Doc changes
Add basement outputs to ZoneCoupledGroundHTBasement.idf

Commit:
Output variable name changes for slab and basement.

mjwitte commented on src/EnergyPlus/PlantPipingSystemsManager.cc:
```diff
@@ -2336,15 +2336,15 @@ namespace PlantPipingSystemsManager {
 			// Zone-coupled slab outputs
-			SetupOutputVariable( "Zone Coupled Surface Heat Flux [W/m2]", PipingSystemDomains( DomainNum ).HeatFlux, "Zone", "Average", PipingSystemDomains( DomainNum ).Name );
-			SetupOutputVariable( "Zone Coupled Surface Temperature [C]", PipingSystemDomains( DomainNum ).ZoneCoupledSurfaceTemp, "Zone", "Average", PipingSystemDomains( DomainNum ).Name );
+			SetupOutputVariable( "GroundDomain:Slab Zone Coupled Surface Heat Flux [W/m2]", PipingSystemDomains( DomainNum ).HeatFlux, "Zone", "Average", PipingSystemDomains( DomainNum ).Name );
```
@mmatts We generally avoid object class'ish names and colons in output variable names.  Better to use "Ground Domain Slab" or "GroundDomain Sab".

mjwitte commented:
@mmatts Since I mixed this up by suggesting output variable name changes, this needs to wait until after v8.4.0 bug fix release, or be split into two separate pull requests.  (Sorry)  Either way, the name changes also need to be added to https://github.com/NREL/EnergyPlus/blob/develop/src/Transition/SupportFiles/Report%20Variables%208-4-0%20to%208-5-0.csv

Commit:
Revert "Output variable name changes for slab and basement."

This reverts commit 74a6a17100d525f53555cd4d3471bf47811be383.

Myoldmopar commented:
@mmatts Here is the contents of the err file in ZoneCoupledGroundHTBasement:

```
Program Version,EnergyPlus, Version 8.4.0-0cdaffec21, YMD=2015.11.18 09:22,IDD_Version 8.4.0
   ** Severe  ** ReadBasementInputs:Site:GroundDomain:Basement="COUPLEDBASEMENT", invalid Basement Floor Boundary Condition Model Name="BASEMENTFLOOROSCM", Condition: Could not match with an Other Side Conditions Model input object.
   ** Severe  ** ReadBasementInputs:Site:GroundDomain:Basement="COUPLEDBASEMENT", invalid Basement Wall Boundary Condition Model Name="BASEMENTWALLOSCM", Condition: Could not match with an Other Side Conditions Model input object.
   ** Severe  ** Invalid Horizontal Insulation Material Name=BASEMENT INSULATION
   **   ~~~   ** Found in BASEMENT INSULATION
   ** Severe  ** Invalid Basement Wall Vertical Insulation Material Name=BASEMENT INSULATION
   **   ~~~   ** Found in BASEMENT INSULATION
   ** Severe  ** PipingSystems:ReadBasementInputs: Perimeter Horizontal Insulation Width is too high compared to basement floor dimensions
  <snip>
```

mitchute commented:
:confused: 
I will look into it...

mjwitte commented:
Probably an order of getinput issue which may be why you had this later in the first place.

Commit:
Moving slab and basement inits to after the environment loop in SetupSimulation.

Commit:
Cleaning up diffs caused by different initialization.

Changed to InitAndSimGroundDomains() SimulateGroundDomains(bool initOnly) and made changes so the domain is not run through an iteration during initialization.

Myoldmopar commented:
@mmatts Am I getting my branches mixed up?  Did a ground-related branch have diffs in the csv files at one point?  This one appears fine, with the eio diffs only being that the lines are in different spots.  But I can't see the numeric diffs in the previous commit either.

mitchute commented:
They were _tiny_ and only for the basement.

Myoldmopar commented:
Aaaah, not crazy.  Well, maybe.  But in either case, this looks good now, no numeric diffs.
